### PR TITLE
Add a README file for Coverity Static analysis

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR adds a REAME file in the `test/Coverity` folder to guide users on the process of building and analyzing the source code using Coverity static analysis tool.

Also, it moves the Coverity config file to the `test/Coverity` folder to be consistent.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/Coverity/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Static code analysis for FreeRTOS-Plus-TCP library
+This directory is made for the purpose of statically testing the MISRA c-2012 compliance of FreeRTOS+TCP using [Synopsys Coverity](https://www.synopsys.com/software-integrity/security-testing/static-analysis-sast.html) static analysis tool. To that end, this directory provides a [CMake](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/test/Coverity/CMakeLists.txt) file and [configuration files](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/main/test/Coverity/ConfigFiles) required to build an application for the tool to analyze.
```
Comment (gitwallit) [inline test/Coverity/README.md:2]: MISRA c -> MISRA C

File: test/Coverity/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+For details regarding the suppressed violations in the report (which can be generated using the following instructions), please see the [MISRA.md](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md) file.
```
Comment (gitwallit) [inline test/Coverity/README.md:4]: which can be generated using the following instructions -> which can be generated using the instructions described below

File: test/Coverity/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Getting Started
+### Prerequisites
+You can run this on any platform which is supported by Coverity. The list and other details can be found [here](https://sig-docs.synopsys.com/polaris/topics/c_coverity-compatible-platforms.html).
```
Comment (gitwallit) [inline test/Coverity/README.md:8]: on any platform which is supported by Coverity->on a platform supported by Coverity

File: test/Coverity/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+To compile and run this project successfully, you must have the following:
+
+1. CMake version > 3.13.0 (You can check whether you have this by typing `cmake --version`)
+    - Not found? Try `apt-get install cmake`
```
Comment (gitwallit) [inline test/Coverity/README.md:12]: Try `apt-get install cmake` -> For Debian based systems, including Ubuntu, try `apt-get install cmake`

File: test/Coverity/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    - Try the `cmake --version` command. If the version number is >= 3.13.0, skip to (4.) or else, continue.
```
Comment (gitwallit) [inline test/Coverity/README.md:13]: Can remove "Try the `cmake --version` command." as we've already mentioned running cmake --version in the line above.

File: test/Coverity/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    - You will need to get the latest CMake version using curl or wget (or similar command).
+        - Uninstall the current version of CMake using `sudo apt remove --purge --auto-remove cmake`.
```
Comment (gitwallit) [inline test/Coverity/README.md:15]: `sudo apt remove --purge --auto-remove cmake` Refer to previous comment about apt availability and Debian based systems.

File: test/Coverity/README.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        - Download the 3.13.0 version using `wget https://cmake.org/files/v3.13/cmake-3.13.0.tar.gz`.
+        - Extract the cmake download using `tar -xzvf cmake-3.13.0.tar.gz`.
+        - Go to the extracted folder (`cd cmake-3.13.0`) and run `./bootstrap`.
+        - Run `make -j$(nproc)' and then run `sudo make install`.
```
Comment (gitwallit) [inline test/Coverity/README.md:19]: `make -j$(nproc)' and then run `sudo make install`. 

-> `make -j$(nproc)` and then run `sudo make install`.
Once you fix the backtick, the lines below that appear as instructions should sort themselves out.

File: test/Coverity/README.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        - Check the version using `cmake --version` command.
+2. GCC compiler
+    - You can see the downloading and installation instructions [here](https://gcc.gnu.org/install/).
+3. Download the repo and include the submodules using the following commands.
+    - `git clone --recurse-submodules https://github.com/FreeRTOS/FreeRTOS-Plus-TCP.git ./FreeRTOS_TCP`
+    - `cd ./FreeRTOS_TCP`
+    - `git submodule update --checkout --init --recursive`
+
+### To build and run coverity:
+Go to the root directory of the FreeRTOS+TCP repo and run the following commands in terminal:
```
Comment (gitwallit) [inline test/Coverity/README.md:29]: FreeRTOS+TCP repo -> FreeRTOS-Plus-TCP repo
For consistency's sake.

File: test/Coverity/README.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  cov-configure --force --compiler cc --comptype gcc
+  ~~~
+2. Create the build files using CMake in a `build` directory
+  ~~~
+  cmake -B build -S test/Coverity
+  ~~~
+3. Go to the build directory and copy the coverity configuration file
+  ~~~
+  cd build/
+  cp ../tools/coverity_misra.config .
```
Comment (gitwallit) [inline test/Coverity/README.md:41]: should we move the coverity_misra.config file under the test/Coverity directory?

File: test/Coverity/README.md (hunk 5 above)
Comment (AniruddhaKanhere) [reply test/Coverity/README.md:13]: This is required as the user must see whether their cmake (which they installed in the step above) is the correct version

**Review by gitwallit — APPROVED** [review]

**Review by gitwallit — APPROVED** [review]

**Review by gitwallit — APPROVED** [review]

File: test/Coverity/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+1. CMake version > 3.13.0 (You can check whether you have this by typing `cmake --version`)
+    - Not found? For Debian based systems, including Ubuntu, try `apt-get install cmake`
+    - Try the `cmake --version` command. If the version number is >= 3.13.0, skip to (4.) or else, continue.
+    - You will need to get the latest CMake version using curl or wget (or similar command).
+        - For Debian based systems, including Ubuntu, uninstall the current version of CMake using `sudo apt remove --purge --auto-remove cmake`.
+        - Download the 3.13.0 version using `wget https://cmake.org/files/v3.13/cmake-3.13.0.tar.gz`.
+        - Extract the cmake download using `tar -xzvf cmake-3.13.0.tar.gz`.
+        - Go to the extracted folder (`cd cmake-3.13.0`) and run `./bootstrap`.
+        - Run `make -j$(nproc)` and then run `sudo make install`.
+        - Check the version using `cmake --version` command.
```
Comment (archigup) [inline test/Coverity/README.md:20]: For Debian based systems, we should recommend using a PPA, or just unpacking it to /opt. Using `sudo make install` makes the system a mess and can cause conflicts with the system package manager

File: test/Coverity/README.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
+# Static code analysis for FreeRTOS-Plus-TCP library
+This directory is made for the purpose of statically testing the MISRA C:2012 compliance of FreeRTOS+TCP using [Synopsys Coverity](https://www.synopsys.com/software-integrity/security-testing/static-analysis-sast.html) static analysis tool. To that end, this directory provides a [CMake](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/test/Coverity/CMakeLists.txt) file and [configuration files](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/main/test/Coverity/ConfigFiles) required to build an application for the tool to analyze.
+
+For details regarding the suppressed violations in the report (which can be generated using the instructions described below), please see the [MISRA.md](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md) file.
+
+## Getting Started
+### Prerequisites
+You can run this on a platform supported by Coverity. The list and other details can be found [here](https://sig-docs.synopsys.com/polaris/topics/c_coverity-compatible-platforms.html).
+To compile and run this project successfully, you must have the following:
```
Comment (archigup) [inline test/Coverity/README.md:9]:
```suggestion
To compile and run the Coverity target successfully, you must have the following:
```

File: test/Coverity/README.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  ~~~
+6. Format the errors in HTML format so that it is more readable while removing the FreeRTOS-Kernel directory from the report
+  ~~~
+  cov-format-errors --dir . --exclude-files '(.*/FreeRTOS-Kernel/.*)' --html-output html-output
+  ~~~
+
+You should now have the HTML formatted violations list in a directory named `html-output`.
+With the current configuration and the provided project, you should see only one deviation from advisory rule 8.13 in file FreeRTOS_IP.c [here](https://github.com/alfred2g/FreeRTOS-Plus-TCP/blob/misra_remaining/source/FreeRTOS_IP.c#L236). This deviation has a justification outlined [here](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-813). With that justification in place, a coverity suppression statement has been added to the code. However, even with that suppression in place, the coverity tool continues to report the deviation. Thus, as an excpetion, we have allowed the deviation to be reported in the HTML formatted report. If you find a way around it, please help us fix this by creating a pull-request in this repository.
```
Comment (archigup) [inline test/Coverity/README.md:58]: This was fixed in a newer version of Coverity, if I recall correctly.

**Review by gitwallit — APPROVED** [review]

**Review by archigup — APPROVED** [review]

**Review by gitwallit — APPROVED** [review]

**Pull request merged**